x-pack/plugins/security_solution/public/common/components/stat_items/stat_items.tsx

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiPanel,
+  EuiHorizontalRule,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import React, { useMemo } from 'react';
+import deepEqual from 'fast-deep-equal';
+
+import { AreaChart } from '../charts/areachart';
+import { BarChart } from '../charts/barchart';
+
+import { histogramDateTimeFormatter } from '../utils';
+
+import { StatItemHeader } from './stat_item_header';
+import { useToggleStatus } from './use_toggle_status';
+import type { StatItemsProps } from './types';
+import { areachartConfigs, barchartConfigs, FlexItem, ChartHeight } from './utils';
+import { Metric } from './metric';
+import { MetricEmbeddable } from './metric_embeddable';
+import { useIsExperimentalFeatureEnabled } from '../../hooks/use_experimental_features';
+import { LensEmbeddable } from '../visualization_actions/lens_embeddable';
+
+export const StatItemsComponent = React.memo<StatItemsProps>(
+  ({
+    areaChart,
+    barChart,
+    description,
+    enableAreaChart,
+    enableBarChart,
+    fields,
+    from,
+    grow,
+    id,
+    loading = false,
+    showInspectButton = true,
+    index,
+    updateDateRange,
+    statKey = 'item',
+    to,
+    barChartLensAttributes,
+    areaChartLensAttributes,
+    setQuerySkip,
+  }) => {
+    const isBarChartDataAvailable = !!(
+      barChart &&
+      barChart.length &&
+      barChart.every((item) => item.value != null && item.value.length > 0)
+    );
+    const isAreaChartDataAvailable = !!(
+      areaChart &&
+      areaChart.length &&
+      areaChart.every((item) => item.value != null && item.value.length > 0)
+    );
+
+    const timerange = useMemo(
+      () => ({
+        from,
+        to,
+      }),
+      [from, to]
+    );
+
+    const { toggleStatus, toggle } = useToggleStatus({ id, setQuerySkip });
+
+    const isChartEmbeddablesEnabled = useIsExperimentalFeatureEnabled('chartEmbeddablesEnabled');
+
+    return (
+      <FlexItem grow={grow} data-test-subj={`stat-${statKey}`}>
+        <EuiPanel hasBorder>
+          <StatItemHeader toggle={toggle} toggleStatus={toggleStatus} description={description} />
```

Comment:
NIT: The `toggleStatus` prop is a bit confusing, it can be read as a function to toggle the status, but it is not, it's the boolean status of the toggle thing, that's because we use the word "toggle" as a name and as a verb in the same place, we should avoid that.
Anyway, could we rename the `toggle` prop to `onToggle` or `onToggleStatus`? This way it is a little bit clearer which one is the callback function?